### PR TITLE
remove generation of redundant package imports

### LINKED ISSUE #101: Redundant package import
Using Smithy `0.12.1`. Found this as the package import being generated is causing compilation errors because of our project structure. Can work around it but I don't think there's a reason for the package import to be added? Doesn't happen in all cases, seems to be if the namespace being imported entirely contains the other it works fine, otherwise the package import is added.

Example:

`errors.smithy`
```
namespace com.example.smithy.errors

@error("client")
@httpError(404)
structure NotFoundError {
  error: String,
}
```
`example.smithy`
```
namespace com.example.smithy.api

service ExampleService {
  operations: [ExampleOperation],
  errors: [com.example.smithy.errors#NotFoundError],
}

@http(method: "GET", uri: "/example", code: 200)
operation ExampleOperation {
  input: ExampleInput,
  output: ExampleOutput,
  errors: [com.example.smithy.errors#NotFoundError],
}

@input
structure ExampleInput {
  @httpQuery("id")
  id: String
}

@output
structure ExampleOutput {
  id: String
}
```

The generated `ExampleService.scala` has a package import for the namespace with the error models:
```
import ExampleServiceGen.ExampleOperationError
import com.example.smithy.errors
import com.example.smithy.errors.NotFoundError
import smithy4s.syntax._

trait ExampleServiceGen[F[_, _, _, _, _]] {
```

### PR DESCRIPTION
**Pull request by lewisjkl**
Closes #101.

### CODE REVIEW

File: modules/codegen/src/smithy4s/codegen/Renderable.scala [hunk 1]
```diff
@@ -43,8 +43,7 @@ object Renderable {
 
   implicit def tupleRenderable[A](implicit
       A: Renderable[A]
-  ): Renderable[(String, A)] = (t: (String, A)) =>
-    A.render(t._2).addImport(t._1)
+  ): Renderable[(String, A)] = (t: (String, A)) => A.render(t._2)
```
Comment (Baccata) [inline modules/codegen/src/smithy4s/codegen/Renderable.scala:46]: I'm a bit suspicious of this. The LHS of the tuple isn't used anymore, which means that this week's implicit is probably useless and its current usesites should be amended

Comment (lewisjkl) [reply modules/codegen/src/smithy4s/codegen/Renderable.scala:46]: That's fair, I will go through and cleanup the usesites

Comment (kubukoz) [reply modules/codegen/src/smithy4s/codegen/Renderable.scala]: > this week's implicit

:joy:

**Review by kubukoz — APPROVED** [review]

File: modules/codegen/src/smithy4s/codegen/Renderable.scala (hunk 1 above)
Comment (Baccata) [reply modules/codegen/src/smithy4s/codegen/Renderable.scala]: Lol, that's what happens when I look at github on the phone whilst being sleep deprived 😄

**Review by Baccata — APPROVED** [review]

**Pull request merged**